src/main/java/kr/codesquad/freddie/chess/board/Board.java

```diff
@@ -0,0 +1,73 @@
+package kr.codesquad.freddie.chess.board;
+
+import kr.codesquad.freddie.chess.piece.Pawn;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Board {
+    // 체스에서 row를 rank라고 한다.
+    public static final int RANK_SIZE = 8;
+    private List<File> files = new ArrayList<>();
+
+    public Board() {
+        for (int i = 0; i < RANK_SIZE; i++) {
+            files.add(new File());
+        }
+    }
+
+    public void add(Pawn pawn) {
+        for (File file : files) {
+            if (file.size() < File.SIZE) {
+                file.add(pawn);
+                return;
+            }
+        }
+
+        if (RANK_SIZE * File.SIZE == size()) {
+            throw new IllegalArgumentException("더 이상 추가할 수 없습니다. 현재 크기 : " + size());
+        }
+    }
```

Comment:
65번째 pawn add할때는 불필요한 포문을 거치게 되는데요, 사이즈 체크하는게 상단으로 오면 어떨까 하네요!

```suggestion
    public void add(Pawn pawn) {
        if (RANK_SIZE * File.SIZE == size()) {
            throw new IllegalArgumentException("더 이상 추가할 수 없습니다. 현재 크기 : " + size());
        }

        for (File file : files) {
            if (file.size() < File.SIZE) {
                file.add(pawn);
                return;
            }
        }
    }
```

Reply:
그리고 `file.size() < File.SIZE` 로 비교하는 로직을 file안으로 집어넣을 수도 있지 않을까? 생각해보시면 좋을것 같습니다.
```java
for (File file : files) {
  file.add(pawn); // 이렇게 하고 갯수를 맞추는 작업은 file 내에서 수행
}
```
또는 
```java
for (File file : file) {
  if (file.isAddable()) { // board가 꼭 file.size 를 알아야 하지 않는거라면 이렇게도 할 수 있겠죠?
    file.add(pawn);
  }
}
```

Reply:
감사합니다!! 완전 놓치고 있었던 부분이네요 🤔
말씀해주신 부분 외에도 for문 중간에 return이 들어가서 흐름이 조금 부자연스럽다고 느껴졌습니다.
스트림을 이용하면 좀 더 깔끔하게 해볼 수 있을 것 같은데 스트림으로 개선해봐도 괜찮을까요?

Reply:
스트림은 앞으로 포문보다 더 자주 사용하게 되실 수도 있으니 이렇게도 저렇게도 바꿔보면서 개선해보세요!

Reply:
수정했습니다.
추가 가능한 file만 필터로 걸러낸 뒤 첫 번째 항목을 찾아 add 해줬습니다.
여기서 만약 추가 가능한 것이 없다면 예외를 발생시키는데, 
이 것도 처음에 말씀해주신 '65번째 pawn add할때는 불필요한 포문을 거치게 되는데요' 처럼,
65번째 add를 할 때 불필요한 스트림을 거치게 되는 것으로 생각해봐야하는 것인지 궁금합니다!

Reply:
음.. 네 사실 그렇져ㅎㅎ 말씀하신 대로 스트림을 활용하면 먼저 사이즈 체크를 하는 방법이 없으니, 이럴땐 포문을 활용하는게 더 좋을 수도 있습니다. 이외에 스트림은 포문과 성능에 대해 비교했을 때 결코 좋거나 하지는 않지만 자바에서 리스트 처리를 함수형 방식으로 짜게 해줄 수 있으니 스트림에 대해 공부를 해보시면 나중에 도움이 많이 되실 거에요. 

그렇지만 자바의 리스트, 맵과 같은 자료 구조에 대해 먼저 학습하시는 것을 권장합니다!

Reply:
감사합니다! 참고하여 학습해보겠습니다.